examples/website/collision-filter/app.jsx

```diff
@@ -0,0 +1,158 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import {createRoot} from 'react-dom/client';
+import {Map} from 'react-map-gl';
+import maplibregl from 'maplibre-gl';
+import {WebMercatorViewport} from '@deck.gl/core';
+import DeckGL from '@deck.gl/react';
+import {GeoJsonLayer, TextLayer} from '@deck.gl/layers';
+import roads from './data/ne_10_roads_mexico.json';
+import {CollisionFilterExtension} from '@deck.gl/extensions';
+import * as turf from '@turf/turf';
+
+const initialViewState = {longitude: -100, latitude: 24, zoom: 5, minZoom: 5, maxZoom: 12};
+const LINE_COLOR = [0, 173, 230];
+
+function calculateLabels(data, pointSpacing) {
+  const routes = data.features.filter(d => d.geometry.type !== 'Point');
+
+  // Add points along the lines
+  const filteredLabels = routes.map(d => {
+    const lineLength = Math.floor(turf.lineDistance(d.geometry));
+
+    const result = {
```

Comment:
There is no reason to create new GeoJSON features if you are not even using a GeoJsonLayer. Just return the generated info with a link back to the source feature.

```ts
{
  label: string;
  anchor: [number, number];
  angle: number;
  priority: number;
  road: Feature;
}[]
```